unicodetools/src/main/java/org/unicode/text/UCD/GenerateData.java

```diff
@@ -811,6 +814,57 @@ public static void writeNormalizerTestSuite(String directory, String fileName)
         for (final String testSuiteCase : testSuiteCases) {
             writeLine(testSuiteCase, log, false);
         }
+        // At least one implementation (ICU4X) has an edge case when a character
+        // whose decomposition contains multiple starters and ends with a
+        // non-starter is followed by a non-starter of lower CCC.
+        // See https://github.com/unicode-org/unicodetools/issues/656
+        // and https://github.com/unicode-org/icu4x/pull/4530.
+        // That implementation also has separate code paths for the BMP and
+        // higher planes.  No such decompositions currently exist outside the
+        // BMP, but by generating these test cases we ensure that this would be
+        // covered.
+        // We stick them in Part 0, which is in principle for handcrafted test
+        // cases, because there are not many of them, and the edge case feels a
+        // tad too weird to describe in the title of a new part.
+        final org.unicode.props.UnicodeProperty sc =
+                IndexUnicodeProperties.make().getProperty(UcdProperty.Script);
+        final org.unicode.props.UnicodeProperty ccc =
+                IndexUnicodeProperties.make().getProperty(UcdProperty.Canonical_Combining_Class);
+        loopOverCodePoints:
+        for (final String cp : UnicodeSet.ALL_CODE_POINTS) {
+            final String[] decompositions =
+                    new String[] {Default.nfd().normalize(cp), Default.nfkd().normalize(cp)};
+            for (final String decomposition : decompositions) {
+                final int lastCCC =
+                        Default.ucd()
+                                .getCombiningClass(
```

Comment:
Nit: Why don't you use ccc instead of Default.ucd().getCombiningClass()? It seems like you would want to use the object for the property if you already got that.

Reply:
I got that, but I am not actually using it, and while `IndexUnicodeProperties` is the more modern thing, it is inconvenient here, as I want an integer, not a string, for the comparisons below. Removed the declaration of `ccc`.
(Someday we bring stronger typing to `IndexUnicodeProperties`.)